libsolidity/formal/BMC.cpp

```diff
@@ -497,11 +497,37 @@ pair<smtutil::Expression, smtutil::Expression> BMC::arithmeticOperation(
 
 	auto values = SMTEncoder::arithmeticOperation(_op, _left, _right, _commonType, _expression);
 
+	IntegerType const* intType = nullptr;
+	if (auto const* type = dynamic_cast<IntegerType const*>(_commonType))
+		intType = type;
+	else
+		intType = TypeProvider::uint256();
+
+	// Mod does not need underflow/overflow checks.
+	// Div only needs overflow check for signed types.
+	if (_op == Token::Mod || (_op == Token::Div && !intType->isSigned()))
+		return values;
+
+	VerificationTarget::Type type;
+	// The order matters here:
+	// If _op is Div, intType is signed, but we only care about overflow.
+	if (_op == Token::Div)
+		type = VerificationTarget::Type::Overflow;
```

Comment:
It's fine as it is, but removing the ``Div`` case above and doing it all here would be easier to read, wouldn't it?
```suggestion
	if (_op == Token::Div)
	{
		if (intType->isSigned())
			type = VerificationTarget::Type::Overflow; // Signed division can (only) overflow.
		else
			return values; // Unsigned division cannot overflow/underflow.
	}
```

Reply:
Yes, the hidden "problem" with that is that `SMTEncoder::arithmeticOperation` adds `right != 0` to `m_context`, and since the current constraints are added to the target in `addVerificationTarget`, the target for `DivByZero` needs to be added before `SMTEncoder::arithmeticOperation` is called.

Reply:
I didn't get that - my suggestion is *exactly* equivalent to the current code, it's just cosmetics...

Reply:
Ah ok, thought you were talking about the stuff before the call to `SMTEncoder::arith...`